### PR TITLE
Added custom cut point in RFM Segmentation

### PR DESCRIPTION
**Pull request by mayurkmmt**
Added custom cut points for rfm_scores in RFMSegmentation and update test cases

## Summary by CodeRabbit

- **New Features**
  - Added support for flexible RFM segmentation, allowing users to specify either the number of bins or custom percentile cut points for recency, frequency, and monetary metrics.

- **Bug Fixes**
  - Improved error handling for invalid segmentation parameters, providing clearer validation and feedback.

- **Tests**
  - Expanded test coverage to include custom segmentation scenarios, parameter validation, edge cases, lazy evaluation, and compatibility checks.

### CODE REVIEW

File: pyretailscience/segmentation/rfm.py [hunk 1]
```diff
@@ -111,27 +170,71 @@ def _compute_rfm(self, df: ibis.Table, current_date: datetime.date) -> ibis.Tabl
...
-        )
-        window_monetary = ibis.window(
-            order_by=[ibis.asc(customer_metrics.monetary), ibis.asc(customer_metrics.customer_id)],
-        )
-
         rfm_scores = customer_metrics.mutate(
-            r_score=(ibis.ntile(10).over(window_recency)),
-            f_score=(ibis.ntile(10).over(window_frequency)),
-            m_score=(ibis.ntile(10).over(window_monetary)),
+            r_score=self._compute_score(customer_metrics, "recency_days", self.r_segments),
```
Comment (mvanwyk) [inline pyretailscience/segmentation/rfm.py:174]: I realized this is a bug. Recency should be sorted descending. The function `_compute_score` need to changed to correctly handle the `ascending` parameter and this needs to be set to `ascending=False` here.

File: pyretailscience/segmentation/rfm.py [hunk 2]
```diff
@@ -111,27 +170,71 @@ def _compute_rfm(self, df: ibis.Table, current_date: datetime.date) -> ibis.Tabl
...
+
+        sorted_segments = sorted(segments)
+        case_expr = ibis.literal(0).cast("int32")
+
+        for i, cutpoint in enumerate(sorted_segments):
+            condition = percentile > ibis.literal(cutpoint)
+            case_expr = condition.ifelse(
+                ibis.literal(i + 1).cast("int32"),
+                case_expr,
+            )
```
Comment (mvanwyk) [inline pyretailscience/segmentation/rfm.py:234]: Could this be simplified with the percent_rank function? https://ibis-project.org/reference/expression-generic#ibis.expr.types.generic.Column.percent_rank

File: tests/segmentation/test_rfm.py [hunk 3]
```diff
@@ -9,6 +10,11 @@
 
 cols = ColumnHelper()
 
+MAX_R = 4
+MAX_F = 2
+MAX_M = 3
+EXPECTED_RESULT_LEN = 20
```
Comment (mvanwyk) [inline tests/segmentation/test_rfm.py:16]: Can you move these into the relevant functions please.
